WordPress/src/main/java/org/wordpress/android/ui/sitecreation/segments/NewSiteCreationSegmentViewHolder.kt

```diff
@@ -35,8 +35,10 @@ sealed class NewSiteCreationSegmentViewHolder(internal val parent: ViewGroup, @L
             title.text = uiState.title
             subtitle.text = uiState.subtitle
             imageManager.load(icon, IMAGE, uiState.iconUrl)
+            // TODO: Check with designers on how to handle this state
+            val iconColor = if (uiState.iconColor.isNotBlank()) uiState.iconColor else R.color.black.toString()
             try {
-                icon.setColorFilter(Color.parseColor(uiState.iconColor), PorterDuff.Mode.SRC_IN)
+                icon.setColorFilter(Color.parseColor(iconColor), PorterDuff.Mode.SRC_IN)
             } catch (e: IllegalArgumentException) {
                 AppLog.e(AppLog.T.SITE_CREATION, "Error parsing segment icon color: ${uiState.iconColor}")
             }
```

Comment:
> The R.color.black.toString() statement converts the black color resource ID int into a String setting the iconColor value to something like "2131099691" rather than getting the text value of the black color reference like "#ff000000" as expected.

I should have checked that, nice catch.

> Rather than setting the default iconColor value programmatically, what about adding the android:tint attribute to the icon view in the new_site_creation_segment_item layout and updating the color filter programmatically if the uiState.iconColor is not blank?

Unfortunately that won't work since this is a `ViewHolder` if it's being reused, it could have the color from the previous use instead of the one set in the layout.

I will fix this with a temporary black color for now and will do a proper fix once we know what we actually want to have happen. We might treat the missing `iconUrl` and `iconColor` as invalid states and there is still some discussion going on about this with our backend friends.

Reply:
> Unfortunately that won't work since this is a `ViewHolder` if it's being reused, it could have the color from the previous use instead of the one set in the layout.

Ahh, you're right.  It seems like the only solution is to set it programmatically then.

> I will fix this with a temporary black color for now and will do a proper fix once we know what we actually want to have happen. We might treat the missing `iconUrl` and `iconColor` as invalid states and there is still some discussion going on about this with our backend friends.

Sounds good.

Reply:
Added the temporary fix: 5553b16c4b7dae6408825f6b873f7c1eb2b12c4c